### ISSUE TITLE
Question: Will Glide use ImageDecoder API of Android P?

### ISSUE THREAD

**Issue by AndroidDeveloperLB** [bumptech/glide#3062]
It might help in decoding some image formats, including even animated GIF/WEBP:
https://www.youtube.com/watch?v=LBBqTd6uOd4&feature=youtu.be&t=116
https://developer.android.com/preview/features?authuser=1#decoding-images
https://developer.android.com/reference/android/graphics/drawable/AnimatedImageDrawable

Not sure what should be done for pre-Android-P though...

According to my tests, at least for GIF, I don't see any advantage (or at least for memory usage) yet of using it , though:
https://stackoverflow.com/q/50104758/878126

**Comment by sjudd** [comment]
Yes, it's on my list of things to do. It's going to be a little tricky to release it publicly because we have to wait for P to be released so that we can at least compile against P in Glide. Every year when we make this change we break a few people's builds.

**Comment by AndroidDeveloperLB** [comment]
@sjudd But do you think it's worth it?
Does it work better in some way than Glide?
Maybe decode better in some way: memory/speed ? 
Starting from Android 8.0, I know that bitmaps are stored in the native memory anyway, so I don't get what's the point of the new API...
Will Glide somehow support animated images (GIF/WEBP ) ? If so, will it be supported on older Android versions?

**Comment by AndroidDeveloperLB** [comment]
@sjudd It seems Google plans to add ImageDecoder as a support library:
https://issuetracker.google.com/issues/78041382

**Comment by AndroidDeveloperLB** [comment]
@stale Why? Don't you think it's important to try it out, to see if it could help for Glide?

**Comment by AndroidDeveloperLB** [comment]
@stale Thank you. Please don't do it again. It's more of a request...

**Comment by AndroidDeveloperLB** [comment]
@stale Please no

**Comment by wbinarytree** [comment]
Hey. I saw the latest master already have the changes for `ImageDecoder`. Any idea which release will ship with it? I see 4.10 is likely not the one.
